src/Docker/Configuration/Container.php

```diff
@@ -79,7 +79,17 @@ public function getConfigTreeBuilder()
                 ->arrayNode('table_files')
         ;
         OutputTableFile::configureNode($outputTableFile);
-
+        // validation of use_only_file_storage with table_files
+        $root->validate()
+            ->ifTrue(function ($v) {
+                if (!empty($v['runtime']['use_file_storage_only']) && empty($v['storage']['output']['table_files'])) {
+                    return true;
+                }
+                return false;
+            })
+            ->thenInvalid('table_files must be specified for use_file_storage_only_mode')
```

Comment:
and must it? like why? what if there is no output?

Reply:
yeah, it must be set, but it can be empty,  let me just double-check that that passes

Reply:
or maybe there shouldn't be any validation required, and the check can go in the implementation.  I'll take another quick look at it

Reply:
> yeah, it must be set

in that case it doesn't have to be set, the entire `storage` node is optional

Reply:
ok, I'll take it out, thanks